integrations/huggingface-transformers/tutorials/question-answering/question_answering-cli.md

```diff
@@ -0,0 +1,372 @@
+dict_to_json_file("squadshifts-train.json", train_dict)
+dict_to_json_file("squadshifts-val.json", val_dict)
+```
+
+#### **Kick off Training**
+
+To use the local files with the CLI, pass `--train_file squadshifts-train.json --validation_file squadshifts-val.json`:
+
+Run the following:
+```bash
+sparseml.transformers.train.question_answering \
+  --output_dir obert_base_pruned90_quant_squadshifts \
+  --recipe zoo:nlp/question_answering/obert-base/pytorch/huggingface/squad/pruned90_quant-none \
+  --recipe_args '{"num_epochs":8, "qat_start_epoch":4.0, "observer_epoch":7.0}' \
+  --model_name_or_path zoo:nlp/masked_language_modeling/obert-base/pytorch/huggingface/wikipedia_bookcorpus/pruned90-none \
+  --distill_teacher dense_tracher \
+  --train_file squadshifts-train.json --validation_file squadshifts-val.json \
+  --do_train --do_eval --evaluation_strategy epoch --logging_strategy epoch --save_steps 1000 --preprocessing_num_workers 32 \
+  --per_device_train_batch_size 8 --per_device_eval_batch_size 32 --gradient_accumulation_steps 2 \
+  --max_seq_length 384 --doc_stride 128 \
+  --seed 42
+```
+
+Without doing any hyperparameter search, the script runs for 13 epochs and converges to ~70% F1 score.
+
+Note that in this case, we used the SQuAD transfer learning recipe (identified by 
+`zoo:nlp/question_answering/obert-base/pytorch/huggingface/squad/pruned90_quant-none`). Since the Squadshifts dataset is similiar to the SQuAD dataset, 
+we chose the same hyperparameters. To adjust the hyperparameters, either download the recipe and modify directly (passing
+a local file to the CLI) or pass an argument to `--recipe_args`.
+
+### Sparse Transfer Learning with a Custom Teacher
+
+To support the transfer learning process, we can apply model distillation, just like we did for the SQuAD case.
+You are free to use the native Hugging Face workflows to train the dense teacher model (and can even
+pass a Hugging Face model identifier to the command), but you can also use the SparseML CLI as well. 
+
+#### Train The Dense Teacher
+
+Run the following to train a dense teacher model on SquadShifts:
+```
+sparseml.transformers.train.text_classification \
+  --output_dir dense-teacher_rotten_tomatoes \
+  --model_name_or_path zoo:nlp/masked_language_modeling/bert-base/pytorch/huggingface/wikipedia_bookcorpus/base-none \
+  --recipe zoo:nlp/sentiment_analysis/bert-base/pytorch/huggingface/sst2/base-none \
+  --recipe_args '{"init_lr":0.00003}' \
+  --dataset_name rotten_tomatoes --input_column_names "text" --label_column_name "label" \
+  --max_seq_length 128 --per_device_train_batch_size 32 --per_device_eval_batch_size 32 --preprocessing_num_workers 6 \
+  --do_train --do_eval --evaluation_strategy epoch --fp16  \
+  --save_strategy epoch --save_total_limit 1
+```
+
+The model converges to ~72% accuracy without any hyperparameter search.
+
+Note that used the dense version of BERT (the stub ends in `base-none`) as the starting point for the training 
```

Comment:
Note that we used the dense version of BERT (the stub ends in base-none) as the starting point for the training and passed a recipe from SparseZoo which was used to train the dense teacher for the SST2 task. Since the SST2 task is similar to the Rotten Tomatoes task, these parameters are a solid place to start. This recipe contains no sparsity-related modifiers and only controls the learning rate and the number of epochs